src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -144,19 +180,62 @@ def map_param_values(
                 f"Unable to replace placeholders in body for the '{self.action}' request of the '{self.collection_name}' collection."
             )
 
+        # format the body based on the content type
+        updated_headers, formatted_body = self.format_body(headers, body)
+
         return SaaSRequestParams(
             method=current_request.method,
             path=path,
-            headers=headers,
+            headers=updated_headers,
             query_params=query_params,
-            json_body=json.loads(body) if body else update_values,
+            body=formatted_body,
         )
 
+    @staticmethod
+    def format_body(
+        headers: Dict[str, Any],
+        body: Optional[str],
+    ) -> Tuple[Dict[str, Any], Optional[str]]:
+        """
+        Builds the appropriately formatted body based on the content type,
+        adding application/json to the headers if a content type is not provided.
+        """
+
+        content_type = next(
+            (
+                value
+                for header, value in headers.items()
+                if header.lower() == "content-type"
+            ),
+            None,
+        )
+
+        if body is None:
+            return headers, None
```

Comment:
We could return this first thing before figuring out content_type.

Reply:
Done!